packages/ckeditor5-widget/theme/widget.css

```diff
@@ -89,3 +89,8 @@
 		transform: translate(-50%);
 	}
 }
+
+.ck-editor__editable .ck-widget {
+	/* Necessary to render properly next to floated objects, e.g. side image case. */
+	display: flow-root;
```

Comment:
Does it affect inline widgets?

Reply:
@niegowski True. Will close that PR and make the change only for ToC.